perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_optional_login_namespace.java

```diff
@@ -0,0 +1,43 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.User;
+import cz.metacentrum.perun.core.api.exceptions.AttributeNotExistsException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeAssignmentException;
+import cz.metacentrum.perun.core.bl.AttributesManagerBl;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.UserVirtualAttributesModuleAbstract;
+
+/**
+ * Attribute, that contains login, only if it is available in the namespace.
+ *
+ * @author Matej Hakoš
+ */
+public class urn_perun_user_attribute_def_virt_optional_login_namespace extends UserVirtualAttributesModuleAbstract {
+	@Override
+	public Attribute getAttributeValue(PerunSessionImpl perunSession, User user, AttributeDefinition attribute) {
+		Attribute attr = new Attribute(attribute);
+		String namespace = AttributesManager.NS_USER_ATTR_DEF+":login-namespace:"+attribute.getFriendlyNameParameter();
+		try {
+			Attribute a = perunSession.getPerunBl().getAttributesManagerBl().getAttribute(perunSession, user, namespace);
+			attr.setValue(a.getValue());
+		} catch (AttributeNotExistsException | WrongAttributeAssignmentException | NullPointerException e) {
+			// It's OK, we just return attribute with value null
+			return attr;
```

Comment:
I think we should log error in this case - AttributeNotExistsException would be thrown if the login-namespace attribute definition does not exist in the system, which I consider important being notified about.

Returning attr here is not necessary as it is returned at the end of the method anyway.

Is catching the NullPointerException really used?

Reply:
The NullPointerException was left by a mistake, when I was creating the tests.